src/ls_server.rs

```diff
@@ -251,34 +252,36 @@ impl LsService {
     }
 
     fn init(&self, id: usize, init: InitializeParams) {
-        let result = InitializeCapabilities {
+        let result = InitializeResult {
             capabilities: ServerCapabilities {
-                textDocumentSync: DocumentSyncKind::Incremental as usize,
-                hoverProvider: true,
-                completionProvider: CompletionOptions {
-                    resolveProvider: true,
-                    triggerCharacters: vec![".".to_string()],
-                },
+                text_document_sync: Some(TextDocumentSyncKind::Incremental),
```

Comment:
@nrc What do you mean, do you want it to serialize to the JSON `"text_document_sync":null` if text_document_sync field is None? That be achieved if we remove the `#[serde(skip_serializing_if="Option::is_none")]` attributes. 

I've brought up the issue of null vs missing properties (undefined in JSON terms) with LSP guys here https://github.com/Microsoft/language-server-protocol/issues/87 and argued that they should be semantically equivalent and the protocol should reflect that.

But serializing to null might be clearer, yeah.